inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/EventMatchingUtil.kt

```diff
@@ -0,0 +1,138 @@
+package com.rakuten.tech.mobile.inappmessaging.runtime.utils
+
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.models.appevents.Event
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.models.messages.Message
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.repositories.CampaignRepository
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.responses.ping.Trigger
+import com.rakuten.tech.mobile.inappmessaging.runtime.data.responses.ping.matchingEventName
+import com.rakuten.tech.mobile.sdkutils.logger.Logger
+
+internal interface EventMatchingUtilType {
+
+    /**
+     * Stores logged event.
+     * Event won't be stored if it is not categorized as persistent event
+     * or there are no campaigns with matching triggers.
+     */
+    fun matchAndStore(event: Event)
+
+    fun matchedEvents(campaign: Message): List<Event>
+
+    fun containsAllMatchedEvents(campaign: Message): Boolean
```

Comment:
nit: add note if this is only for testing.

Reply:
`containsAllMatchedEvents` is used by MessageEventReconcilationUtil for campaign validation (not only in tests)